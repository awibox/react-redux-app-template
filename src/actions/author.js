import axios from 'axios';
import {GET_AUTHOR_SUCCESS} from '../actions/types';
import {getError} from "./error";
import {gitHubApiUrl, authorAccount} from '../../configs/config';

const END_POINT = 'users';

const getAuthorSuccess = author => ({
    type: GET_AUTHOR_SUCCESS,
    payload: author
});

export const getAuthor = () => dispatch => {
    axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}`)
        .then(response => {
            const {data} = response;
            dispatch(getAuthorSuccess(data));
        }).catch(err => {
            dispatch(getError(err.response.data))
        });
};
