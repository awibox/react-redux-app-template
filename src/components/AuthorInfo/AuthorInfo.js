import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
// Components
import { Button, Card } from 'tigerspack';
// Styles
import styles from './AuthorInfo.scss';

export default class AuthorInfo extends Component {
  static propTypes = {
    author: ImmutablePropTypes.contains({
      avatar_url: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      company: PropTypes.string.isRequired,
      bio: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired,
    }),
  };

  buttonClick = (url) => window.open(url);

  render() {
    const { author } = this.props;
    return (
      <Card theme={'light'} className={styles.authorInfo}>
        <div className={styles.logo} style={{ backgroundImage: `url(${author.get('avatar_url')})` }} />
        <div className={styles.title}>{author.get('name')}</div>
        <div className={styles.titleVersion}>{author.get('location')}</div>
        <div className={styles.infoText}>{author.get('bio')}</div>
        <div className={styles.button}>
          <Button icon={<i className="fa fa-github"/>} onClick={() => this.buttonClick(author.get('html_url'))}>
            Github page
          </Button>
        </div>
      </Card>
    );
  }
}
