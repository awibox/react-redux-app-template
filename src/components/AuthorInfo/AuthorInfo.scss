.authorInfo {
  text-align: center;
  display: flex;
  flex-flow: column;
  justify-content: center;
}

.logo {
  width: 150px;
  height: 150px;
  background-size: cover;
  background-repeat: no-repeat;
  margin: 30px auto;
}

.title {
  line-height: 56px;
  font-size: 56px;
  font-weight: 100;
  vertical-align: top;
  padding-bottom: 10px;
}

.titleVersion {
  font-size: 18px;
  line-height: 22px;
  margin-left: 5px;
}

.infoText {
  padding: 20px;
  font-size: 20px;
  color: #333;
  font-weight: 100;
  line-height: 30px;
  margin-bottom: 10px;
}

.button {
  padding-bottom: 20px;
}
