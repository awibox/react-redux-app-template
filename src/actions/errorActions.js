import { GET_ERRORS } from 'actions/types';

export const getError = (error) => ({
  type: GET_ERRORS,
  payload: error,
});

export default getError;
