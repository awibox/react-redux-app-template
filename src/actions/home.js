import axios from 'axios';
import {GET_HOME_SUCCESS} from 'actions/types';
import {gitHubApiUrl, authorAccount, repoName} from 'config';
import {getError} from './error';

const END_POINT = 'repos';


export const getHomeSuccess = (home) => ({
  type: GET_HOME_SUCCESS,
  home,
});

export const getHome = () => async (dispatch) => {
  try {
    const {data} = await axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}/${repoName}`);
    dispatch(getHomeSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
