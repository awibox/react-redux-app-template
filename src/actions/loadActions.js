import { GET_LOAD_STATUS } from 'actions/types';

export const setLoadStatus = (isLoadingStatus) => ({
  type: GET_LOAD_STATUS,
  payload: { isLoading: isLoadingStatus },
});

export default setLoadStatus;
