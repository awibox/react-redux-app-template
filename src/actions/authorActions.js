import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';
import { GET_AUTHOR_SUCCESS } from 'actions/types';
import { gitHubApiUrl, authorAccount } from 'config';
import { getError } from 'actions/errorActions';


const END_POINT = 'users';

export const getAuthorSuccess = (author) => ({
  type: GET_AUTHOR_SUCCESS,
  payload: author,
});

export const getAuthorAction = () => async (dispatch) => {
  try {
    const { data } = await trackPromise(axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}`));
    dispatch(getAuthorSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
