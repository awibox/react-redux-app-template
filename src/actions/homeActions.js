import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';
import { GET_HOME_SUCCESS } from 'actions/types';
import { gitHubApiUrl, authorAccount, repoName } from 'config';
import { getError } from './errorActions';

const END_POINT = 'repos';

export const getHomeSuccess = (home) => ({
  type: GET_HOME_SUCCESS,
  payload: home,
});

export const getHomeAction = () => async (dispatch) => {
  try {
    const { data } = await trackPromise(axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}/${repoName}`));
    dispatch(getHomeSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
