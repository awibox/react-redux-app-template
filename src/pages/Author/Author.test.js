import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import AuthorInfo from 'components/AuthorInfo/AuthorInfo';
import { Alert } from 'tigerspack';
import { Map } from 'immutable';
import { getAuthorSuccess } from 'actions/authorActions';
import AuthorContainer from './Author';

describe('Author', () => {
  const mockData = Map({
    avatar_url: 'http://localhost:8888/avatar',
    name: 'John',
    location: 'USA',
    company: 'Google',
    bio: 'Addition information',
    html_url: 'http://localhost:8888/url',
  });

  const initialState = {
    authorState: {
      author: mockData,
    },
    errors: {
      message: 'Fake error',
    },
    load: {
      isLoading: false,
    },
    getAuthorAction: () => {},
  };
  const mockStore = configureStore([thunk]);
  let store; let
    container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = mount(<Provider store={store}>
      <AuthorContainer getAuthorAction={initialState.getAuthorAction} />
    </Provider>);
  });
  it('should render coorectly', () => {
    expect(container).toMatchSnapshot();
  });
  it('should render the container', () => {
    expect(container.length).toEqual(1);
  });
  it('should matches with initialState in the AuthorInfo', () => {
    expect(container.find(AuthorInfo).prop('author')).toEqual(mockData);
  });
  it('should display Alert', () => {
    expect(container.find(Alert)).toExist();
  });
  it('should check prop Alert', () => {
    expect(container.find(Alert).prop('children')).toEqual(initialState.errors.message);
  });
  it('should action on dispatching', () => {
    store.dispatch(getAuthorSuccess(mockData));
    const action = store.getActions();
    expect(action[0].type).toBe('GET_AUTHOR_SUCCESS');
    expect(action[0].payload).toEqual(mockData);
    container.unmount();
  });
});
