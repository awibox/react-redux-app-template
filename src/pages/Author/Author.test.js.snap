// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author should render coorectly 1`] = `
<Connect(AuthorContainer)
  getAuthorAction={[Function]}
>
  <div>
    <ForwardRef(render)
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
      css={
        Object {
          "background": "#D15130",
          "border": "1px solid #D15130",
          "borderRadius": "4px",
          "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24)",
          "color": "#fff",
          "marginBottom": "15px",
          "padding": "15px",
          "position": "relative",
          "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
        }
      }
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="css-1nrs082"
        style={
          Object {
            "display": "block",
          }
        }
      >
        Fake error
      </div>
    </ForwardRef(render)>
    <ForwardRef(render)
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
      className="authorInfo"
      css={
        Object {
          "background": "#fff",
          "border": "1px solid #E5E8EC",
          "borderRadius": "3px",
          "borderTop": undefined,
          "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "marginBottom": "15px",
          "overflow": "hidden",
          "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
        }
      }
    >
      <div
        className="authorInfo css-1an3q4h"
      >
        <ForwardRef(render)
          __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
          css={
            Object {
              "padding": "15px",
            }
          }
        >
          <div
            className="css-1v1qs4"
          >
            <div
              className="logo"
              style={
                Object {
                  "backgroundImage": "url(http://localhost:8888/avatar)",
                }
              }
            />
            <div
              className="title"
            >
              John
            </div>
            <div
              className="titleVersion"
            >
              USA
            </div>
            <div
              className="infoText"
            >
              Addition information
            </div>
            <div
              className="button"
            >
              <ForwardRef(render)
                __EMOTION_TYPE_PLEASE_DO_NOT_USE__="button"
                css={
                  Object {
                    ":disabled": Object {
                      ":hover": Object {
                        "backgroundColor": "#3F88C5",
                        "color": "#fff",
                      },
                      "cursor": "not-allowed",
                      "opacity": 0.7,
                    },
                    ":hover": Object {
                      "backgroundColor": "#619DCF",
                      "color": "#fff",
                      "textDecoration": "none",
                    },
                    "backgroundColor": "#3F88C5",
                    "backgroundImage": "none",
                    "border": "1px solid transparent",
                    "borderRadius": "4px",
                    "boxSizing": "border-box",
                    "color": "#fff",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontSize": "14px",
                    "fontWeight": 500,
                    "lineHeight": 1.52857143,
                    "marginBottom": 0,
                    "outline": "none",
                    "padding": "6px 16px",
                    "position": "relative",
                    "textAlign": "center",
                    "transition": "all 0.5s",
                    "userSelect": "none",
                    "verticalAlign": "middle",
                    "whiteSpace": "nowrap",
                  }
                }
                onClick={[Function]}
              >
                <button
                  className="css-1mjylde"
                  onClick={[Function]}
                >
                  <ForwardRef(render)
                    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
                    css={
                      Object {
                        "marginLeft": "-4px",
                        "marginRight": "8px",
                      }
                    }
                  >
                    <span
                      className="css-1dl6nox"
                    >
                      <i
                        className="fa fa-github"
                      />
                    </span>
                  </ForwardRef(render)>
                  Github page
                </button>
              </ForwardRef(render)>
            </div>
          </div>
        </ForwardRef(render)>
      </div>
    </ForwardRef(render)>
    <div
      style={
        Object {
          "backfaceVisibility": "hidden",
          "background": "rgba(255,255,255,.5)",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 10,
        }
      }
    >
      <div
        style={
          Object {
            "height": 100,
            "left": "50%",
            "position": "absolute",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": 100,
          }
        }
      >
        <svg
          className="lds-spinner"
          height="100%"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            key="rotate(0 50 50)"
            transform="rotate(0 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.9166666666666666s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(30 50 50)"
            transform="rotate(30 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.8333333333333334s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(60 50 50)"
            transform="rotate(60 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.75s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(90 50 50)"
            transform="rotate(90 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.6666666666666666s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(120 50 50)"
            transform="rotate(120 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.5833333333333334s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(150 50 50)"
            transform="rotate(150 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.5s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(180 50 50)"
            transform="rotate(180 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.4166666666666667s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(210 50 50)"
            transform="rotate(210 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.3333333333333333s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(240 50 50)"
            transform="rotate(240 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.25s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(270 50 50)"
            transform="rotate(270 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.16666666666666666s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(300 50 50)"
            transform="rotate(300 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="-0.08333333333333333s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
          <g
            key="rotate(330 50 50)"
            transform="rotate(330 50 50)"
          >
            <rect
              fill="#3F88C5"
              height="12"
              rx="9.4"
              ry="4.8"
              width="6"
              x="47"
              y="24"
            >
              <animate
                attributeName="opacity"
                begin="0s"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="1;0"
              />
            </rect>
          </g>
        </svg>
      </div>
    </div>
  </div>
</Connect(AuthorContainer)>
`;
